### ISSUE TITLE
Getting crash

### ISSUE THREAD

**Issue by Raviojha62** [Minitour/EasyNotificationBadge#3]
I was using it for add badge on barbuttonitem . but its not working 
I am getting crash on the 
fatal error: unexpectedly found nil while unwrapping an Optional value

**Comment by Minitour** [comment]
@Raviojha62 Hi, can you include code snippet? And where is the function being called? Although I haven't mentioned it, you need to called the badge function after the view was loaded, so in case you're doing it in a ViewController, you should call it in viewDidAppear and not viewDidLoad.

**Comment by Minitour** [comment]
@Raviojha62 were you able to resolve the issue?

**Comment by keithics** [comment]
Im having the same problem..



    class BaseViewController: UIViewController {
    
    override func viewDidAppear(_ animated: Bool) {
        self.navigationController?.isNavigationBarHidden = false
        
        let rightButton : UIBarButtonItem = UIBarButtonItem(title: "\u{f291}", style: UIBarButtonItemStyle.plain, target: self, action: #selector(showSettings))
        
        rightButton.setTitleTextAttributes([
            NSFontAttributeName : UIFont(name: "FontAwesome", size: 22)!,
            NSForegroundColorAttributeName : UIColor.white],
                                          for: UIControlState())
        
        rightButton.badge(text: "7")
        self.navigationItem.rightBarButtonItem = rightButton
        self.navigationController?.navigationBar.titleTextAttributes = [NSForegroundColorAttributeName: UIColor.white, NSFontAttributeName: UIFont(name: "RNSMiles-Regular", size: 22)!]
        
    }
    
    
    func showSettings(){
        let storyboard = UIStoryboard(name: "Main", bundle: nil)
        let controller = storyboard.instantiateViewController(withIdentifier: "CartViewController")        
        self.present(controller, animated: true, completion: nil)
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(true)
        navigationItem.backBarButtonItem = UIBarButtonItem(title: "", style: .plain, target: nil, action: nil)
    }
    
    }

**Comment by Minitour** [comment]
@keithics Ok so here is the thing; You are setting the badge text before the badge's view was initialized. try replacing:
```swift
    rightButton.badge(text: "7")
    self.navigationItem.rightBarButtonItem = rightButton
```

with:
```swift
    self.navigationItem.rightBarButtonItem = rightButton
    rightButton.badge(text: "7")
```

Since this is an extension and not a sub-class I cannot create a property to store the value until the view is loaded, thus you need to invoke this method after the view was loaded. 

I will be applying a fix to the crash and simply print a message letting you know that you have tried to set a badge on a view that was not loaded. Thanks for contributing.

**Comment by keithics** [comment]
got it! thank you. 👍

**Comment by Minitour** [comment]
Fixed with 1.0.3 release.
